### PR TITLE
Perform substitutions in geany.desktop file

### PR DESCRIPTION
**Pull request by codebrainz**
Keeps this info up to date automatically.

The only part I'm not sure is better is to use the full absolute path in the `Exec` command, though according to the [Desktop Entry Specification](http://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables), it's allowed. I did it because I don't want the Geany in path for my menu, but rather the one that's read from Desktop Entry files in `~/.local` which is where the Geany I want to use in desktop main menu is also located.

Also, I'm not sure of the `_Name` key is translated, but I think it should still work fine since Autoconf processes the file before Intltool.

### CODE REVIEW

**Comment by codebrainz** [comment]
I don't know how to update Waf for this, @eht16 any hints?

File: geany.desktop.in.in [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Type=Application
-Version=1.0
-_Name=Geany
+Version=@VERSION@
```
Comment (b4n) [inline geany.desktop.in.in]: This is not the app's version: http://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#key-version

**Comment by b4n** [comment]
> The only part I'm not sure is better is to use the full absolute path in the `Exec` command, though according to the [Desktop Entry Specification](http://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables), it's allowed. I did it because I don't want the Geany in path for my menu, but rather the one that's read from Desktop Entry files in `~/.local` which is where the Geany I want to use in desktop main menu is also located.

Actually as the `Version` doesn't seem to be the app's version but the version of the standard the file conforms with, I guess the path change is the only meaningful thing left :)

And I don't mind much about the path.  On one hand it's nice to have it run the actually installed executable no matter the PATH, but OTOH expanding `bindir` is a bit of a hack, and not many `.desktops` seem to use full paths (but some do, 89 out of 500 in my */usr/share/applications/*).

> Also, I'm not sure of the `_Name` key is translated, but I think it should still work fine since Autoconf processes the file before Intltool.

Yeah, `_Name` is translated, but it probably isn't a problem as you say, intltool should be just fine.

**Comment by codebrainz** [comment]
@b4n removed the Version replacement, thanks for noticing.

**Comment by eht16** [comment]
The Waf part seems ok.
I wonder why we want to translate "Geany"? It's a name not something which should have been to be translated, I'd say.

@frlan ?

**Comment by sardemff7** [comment]
To substitute `*dir` variables in a file, the *defacto* standard is to use `sed` (well, `$(SED)` of course) and not Autoconf substitution. See the [Autoconf manual](http://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf.html#Installation-Directory-Variables) for details. Quick quote:
>Similarly, you should not rely on `AC_CONFIG_FILES` to replace `bindir` and friends in your shell scripts and other files; instead, let `make` manage their replacement.

So please, use `$(SED)`. :-)

**Comment by codebrainz** [comment]
@sardemff7 can you give the code I should add? I'm not good with Autotool's weird rules about not using the built-in substitution mechanism and instead re-inventing it manually using sed. Also I have no idea how to write sed commands portably.

**Comment by codebrainz** [comment]
I guess something like that works. Is it worth it?

**Comment by frlan** [comment]
I wouldn't translate 'Geany' also. I remember the issues with one of these Linux distributions they did and I really was a mess

**Comment by eht16** [comment]
Re-reading the diff makes me noticing that the translation discussion is completely unrelated to the changes here as the Name field was already translatable before because of the underscore in the field name.

The file wscript needs to be updated in line 574: replace `GEANY_BINDIR` with `bindir`.
Otherwise this looks good to me, the generated geany.desktop contains the replaced and translated contents, both when built with Waf as well as with autotools.

**Comment by eht16** [comment]
@codebrainz if you could remove the wscript changes, we could essentially merge this PR if still wanted.

**Comment by codebrainz** [comment]
Closing since I'm no longer working on this. It needs the waf stuff removed at the least. I'll re-open if I want to continue working on it.

**Pull request closed without merging**